linera-core/src/updater.rs

```diff
@@ -282,14 +282,9 @@ where
                     // For `BlobsNotFound`, we assume that the local node should already be
                     // updated with the needed blobs, so sending the chain information about the
                     // certificates that last used the blobs to the validator node should be enough.
-                    let missing_blob_ids = stream::iter(mem::take(&mut blob_ids))
-                        .filter(|blob_id| {
-                            let node = self.remote_node.node.clone();
-                            let blob_id = *blob_id;
-                            async move { node.blob_last_used_by(blob_id).await.is_err() }
-                        })
-                        .collect::<Vec<_>>()
-                        .await;
+                    let blob_ids = mem::take(&mut blob_ids).into_iter().collect::<Vec<_>>();
```

Comment:
Saving one allocation:

```suggestion
                    let blob_ids = blob_ids.drain().collect::<Vec<_>>();
```

Reply:
Thank you.